src/encoded/static/components/static-pages/placeholders/BasicCarousel.js

```diff
@@ -108,18 +122,38 @@ export class BasicCarousel extends React.PureComponent {
             carouselProps.renderBottomLeftControls = this.renderBottomLeftControls;
         }
 
-        //style
-        //const style = { 'width': slideWidth, 'height': slideHeight };
-        const style = {};
+        let commonImageStyle = null;
+
+        if (typeof autoplay === "boolean") {
+            carouselProps.autoplay = autoplay;
+        }
+
+        if (typeof slideHeight === "number" && typeof slideWidth === "number") {
+            // I forgot if Nuka Carousel can automatically resize itself without presence of slideHeight/slideWidth, but think so.
+            // If we set the below values, then will see multiple slides at once (e.g. current one, + slices of previous/next)
+            // SEE: https://gyazo.com/35f4efc58d428bc18a335bbe3f1228e7
+            // carouselProps.slideHeight = slideHeight + "px"; // Nuka Carousel requires "px" appended and treats floats as percentages, unlike React elements.
+            // carouselProps.slideWidth = slideWidth + "px";
+
+            // Instead we set initialSlideWidth and initialSlideHeight which keeps the 1 current slide visible and others hidden.
+            // Nuka Carousel I believe might then resize slides if needed (?) idk. Or atleast stretch slide containers to be
+            // 100% width of parent container or something.
+            carouselProps.initialSlideHeight = 720;
+            carouselProps.initialSlideWidth = 540;
```

Comment:
```
carouselProps.initialSlideHeight = 720;
carouselProps.initialSlideWidth = 540;
```
height/width is hard-coded. do we assume the carousel slide images' size are always 540x720?

Reply:
Ah, sorry! Is typo I think.

I think in general it would be a consistent image size (among all images in carousel) in _most_ cases but there's possibility they could differ, which partly figured would be decent reason to have heightMode: max.

1 problem with heightMode is that all images need to be loaded first and then like, "analyzed", in JS to get measurements and this might take a while (after init, when slide to) so having an already-known dimension for the image prior to image being loaded is (in general) almost always preferable.

Reply:
Thinking about having an optional "height" and "width" in `props.slides[]` in addition to slideHeight (which by default would apply to all slides unless overridden)? Then we could have ... `<img style={{ width : slide.width || slideWidth || null }}/>`